elastic/beats
# New metricbeat module: pgbouncer

- WHAT:
The module follows the standard design patterns of Metricbeat modules, ensuring consistency with other database modules like PostgreSQL. It implements the MetricSet interface to collect metrics from PgBouncer.
The module interacts with PgBouncer using SQL queries to gather pool statistics and server performance data. The data is parsed and processed through Metricbeat's internal processing pipeline. It communicates with PgBouncer using the native Postgres protocol over TCP. It sends queries to fetch relevant metrics and processes the responses, converting them into Metricbeat-compatible events.
- WHY:
PGBouncer is widely used as a connection pooler for PostgreSQL, and monitoring its performance is crucial for maintaining the health of database-backed systems. By adding this module, Metricbeat provides users with out-of-the-box monitoring capabilities for PGBouncer, enabling them to track metrics such as connection pool utilization, server performance, and query rates. This integration helps users maintain optimal database performance, reduce latency, and prevent resource exhaustion in their systems.
-->

## Checklist




## Disruptive User Impact
It shouldn't impact users.
## Author's Checklist
- [ ] Code follows the established coding guidelines and style.
- [ ] Proper error handling is in place.
- [ ] Tests passing and the coverage > 80%.
- [ ] Code builds as expected.
- [ ] Data collection is efficient, and no unnecessary queries are run.
- [ ] Documentation is updated to reflect the new module, including how to configure and use it.

## How to test this PR locally
1. Run the docker-compose stack and wait until you will be able to reach kibana on `localhost:5601`
```
docker-compose -f ./metricbeat/module/pgbouncer/docker-compose.yml up -d
```
2. Run the metricbeat agent:
```
go run ./metricbeat/main.go -c ./metricbeat/module/pgbouncer/_meta/config_local.yml
```
3. Go to kibana panel, create Data View with the `test` pattern and check logs in discover.
